src/launchCommand.ts

```diff
@@ -59,23 +69,63 @@ export async function addMoreHelpfulVMArgs(config: vscode.DebugConfiguration) {
     }
 }
 
-function checkJavaVersion(javaExec: string): Promise<number> {
-    return new Promise((resolve, _reject) => {
-        cp.execFile(javaExec, ["-version"], {}, (_error, _stdout, stderr) => {
-            const javaVersion = parseMajorVersion(stderr);
-            resolve(javaVersion);
-        });
-    });
+/**
+ * Returns the target runtime version. If the javaExec is not specified, then return the current Java version
+ * that the Java tooling used.
+ * @param javaExec the path of the Java executable
+ */
+export async function getJavaVersion(javaExec: string): Promise<number> {
+    javaExec = javaExec || path.join(await getJavaHome(), "bin", "java");

```

Comment:
We don't have to differ `java.exe` vs `java` by platforms?

Reply:
yes, running "java" and "java.exe" in shell both works in windows.